src/common/popupcmn.cpp

```diff
@@ -431,8 +431,8 @@ bool wxPopupTransientWindow::Show( bool show )
 
 void wxPopupTransientWindow::Dismiss()
 {
-    Hide();
     PopHandlers();
+    Hide();
```

Comment:
I'm not really comfortable with changing this, it risks breaking the other ports.

It looks like the current code dates from 414f2513fc (For wxMSW capture and release the mouse as the cursor moves out or back in to the transient popup window to enable the transient to be automatically dismissed when a click happens outside of it., 2005-04-15) so at the very least this would need to be tested in the popup sample under MSW to check if it doesn't result in any regressions.

Reply:
I just tested `wxMSW` and `wxGTK` with this change and didn't notice any regression. everything work well.

Looking closer at 414f251, I suspect that changing the order of calls in `wxPopupTransientWindow::Dismiss()` was **not** intentional because the role of `wxPopupWindowHandler` and `wxPopupFocusHandler` is to intercept events that cause the popup to disappear. IOW, they have already done their duty if we are really in `Dismiss()`.